src/loader/settings.rs

```diff
@@ -36,6 +40,24 @@ pub struct ReturnInfosToCopy<'a> {
     pub dest: &'a ReturnInfo,
 }
 
+/// Indicates whether a Wasm input is a constant, and if so, its value.
+///
+/// The `must_collapse` cell can be mutated to indicate that the constant
+/// should be collapsed into the instruction, if the ISA supports it.
+pub enum MaybeConstant {
+    NonConstant,
+    CollapsedConstant(WasmValue),
```

Comment:
can you add comments to `Collapsed` and `Reference` variants? what's their difference/purpose? (the `must_collapse` comment could be moved to the variant)

Reply:
Done.